thread1:
  core: 2

thread2:
  core: 2

thread3:
  core: 2